### ISSUE TITLE
Deoplyed to real world? How far are we from AGI?

### ISSUE THREAD

**Issue by yhyu13** [MineDojo/Voyager#30]
1, Maybe a stupid question is skill library the deterministic factor that excel Voyager framework?
2, We can replace GPT4 with any other LLM? What least required specific skills should that LLM have? Meaning what field should that LLM be especially trained on (coding, reasoning)
3, How is minecraft's reward feed back loop compare to real-world environment, is there any research on that? Like, killing a pig and heal is not simply two java api call. What if the feedback long, hard and negative, and more importantly, partially informative.

**Comment by wangguanzhi** [comment]
Thanks a lot for your interest!
1. Voyager consists of three key components: an automatic curriculum for maximizing exploration, an iterative prompting mechanism for improving programs, and a skill library for skill retrieval. Each part is very important for Voyager to work well. Feel free to check out the ablation studies (Sec. 3.4) in the [paper](https://voyager.minedojo.org/assets/documents/voyager.pdf).
2. Both reasoning and coding abilities are essential. We tried to replace GPT-4 with GPT-3.5 but the results were much worse. We did not find any existing LLM that can perform on par with GPT-4.
3. Interesting idea! Minecraft is 3D open-ended game but it's still quite different from a real-world environment. Regarding the feedback, we found that GPT-4 was really good at understanding different types of feedback to improve programs. Even though there are still some limitations with GPT-4 (see Sec. 4 in the paper), we envision that future LLMs would progressively address the limitations.

There is still a long way to go before AGI but we hope that Voyager could give important insights to the community!
